File: Python/simple-factory/simple03/Nvwa.py
# -*- coding: utf-8 -*-
from PersonFactory import PersonFactory

def start():
    try:
        person = PersonFactory.producePerson('MAN')
        person.sameOperation()
        person.diffOperation()

        print

        person = PersonFactory.producePerson('Woman')
        person.sameOperation()
        person.diffOperation()

        print

        person = PersonFactory.producePerson('RObot')
        person.sameOperation()
        person.diffOperation()
    except:
        pass

if __name__ == '__main__':
    start()

File: Homework/Simple Factory/ch4/Nvwa/Nvwa.java
package ch4.Nvwa;

public class Nvwa {

	public static void makePerson(String p){
		Person person = PersonFactory.produceperson(p);
		person.diffOperation();
	}

	public static void main(String[] args) {
        makePerson("M");
		makePerson("W");
        makePerson("R");
	}
}
